neo/Consensus/ConsensusService.cs

```diff
@@ -286,7 +311,17 @@ private void OnPersistCompleted(Block block)
 
         private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage message)
         {
-            if (message.ViewNumber < context.ViewNumber) return;
+            if (message.ViewNumber < context.ViewNumber)
+            {
+                if (!knownHashes.Add(payload.Hash)) return;
+                // If we see a recovery message from a lower view we should recover them to our higher view; they may
+                // be stuck in commit.
+                if (!IsRecoveryAllowed(payload.ValidatorIndex, 1, context.F() + 1 ))
```

Comment:
Double check when ready because maybe this guy `+1` is not in the View, thus, anyway would recover the guy.

Reply:
At least one of `F+1` nodes must be in the correct view or they never would have changed view.